packages/components/src/range-control/types.ts

```diff
@@ -0,0 +1,288 @@
+		/**
+		 * If this property is true, a button to reset the the slider is
+		 * rendered.
+		 *
+		 * @default false
+		 */
+		allowReset?: boolean;
+		/**
+		 * If this property is added, a DashIcon component will be rendered
+		 * before the slider with the icon equal to beforeIcon.
+		 *
+		 * For more information on `IconType` see the Icon component:
+		 * /packages/components/src/icon/index.tsx
+		 */
+		beforeIcon?: IconType< IconProps >;
+		/**
+		 * CSS color string for the `RangeControl` wrapper.
+		 *
+		 * @default COLORS.ui.theme
+		 * @see /packages/components/src/utils/colors-values.js
+		 */
+		color?: CSSProperties[ 'color' ];
+		/**
+		 * The current input to use as a fallback if `value` is currently
+		 * `undefined`.
+		 */
+		currentInput?: number;
+		/**
+		 * An icon to be shown above the slider next to it's container title.
+		 */
+		icon?: string;
+		/**
+		 * If no value exists this prop contains the slider starting position.
+		 */
+		initialPosition?: number;
+		/**
+		 * Passed as a prop to the `NumberControl` component and is only
+		 * applicable if `withInputField` is true. If true, while the number
+		 * input has focus, pressing `UP` or `DOWN` along with the `SHIFT` key
+		 * will change the value by the `shiftStep` value.
+		 *
+		 * @default true
+		 */
+		isShiftStepEnabled?: boolean;
+		/**
+		 * If this property is added, a label will be generated using label
+		 * property as the content.
+		 */
+		label?: string;
+		/**
+		 * Callback for when `RangeControl` input loses focus.
+		 *
+		 * @default () => void
+		 */
+		onBlur?: FocusEventHandler< HTMLInputElement >;
+		/**
+		 * A function that receives the new value. The value will be less than
+		 * `max` and more than `min` unless a reset (enabled by `allowReset`)
+		 * has occurred. In which case the value will be either that of
+		 * `resetFallbackValue` if it has been specified or otherwise
+		 * `undefined`.
+		 *
+		 * @default () => void
+		 */
+		onChange?: ( value?: number ) => void;
+		/**
+		 * Callback for when `RangeControl` input gains focus.
+		 *
+		 * @default () => void
+		 */
+		onFocus?: FocusEventHandler< HTMLInputElement >;
+		/**
+		 * Callback for when mouse exits the `RangeControl`.
+		 *
+		 * @default () => void
+		 */
+		onMouseLeave?: MouseEventHandler< HTMLInputElement >;
+		/**
+		 * Callback for when mouse moves within the `RangeControl`.
+		 *
+		 * @default () => void
+		 */
+		onMouseMove?: MouseEventHandler< HTMLInputElement >;
+		/**
+		 * CSS color string to customize the rail element's background.
+		 */
+		railColor?: CSSProperties[ 'color' ];
+		/**
+		 * A way to customize the rendered UI of the value.
+		 *
+		 * @default ( value ) => value
+		 */
+		renderTooltipContent?: (
+			value?: ControlledRangeValue
+		) => string | number | null | undefined;
+		/**
+		 * The value to revert to if the Reset button is clicked (enabled by
+		 * `allowReset`)
+		 */
+		resetFallbackValue?: number | string | null;
```

Comment:
Since we established that (at least for now) `value` should be of type `number`, should we remove `string` and `null`  from the possible types of `resetFallbackValue` ?

```suggestion
		resetFallbackValue?: number;
```

(this change also improves the Storybook control from this prop, as it changes from "Set object" to "Set number")